Enhancement/9/add cookbock for forms and wait (#33)
sakuli/docs: content/cookbook/handling-forms.md

+++
title = "Handling forms"
date =  2020-03-20T13:18:11+01:00
weight = 5
+++

# Handling forms

> Filling forms and verifying their functionality is one of the major use cases in E2E-Testing and -Monitoring. Most of the web element have been abstracted from their technical specification in HTML. A `<intput name="last-name" /> ` for example is abstracted as a via the `_textbox("last-name")` accessor. Let's have a look at some more of the most common accessors to interact with forms.

Reviewer: ```suggestion
Filling forms and verifying their functionality is one of the major use cases in E2E-Testing and -Monitoring. Most of the web elements have been abstracted from their technical specification in HTML. An `<intput name="last-name" /> ` for example is abstracted as a `_textbox("last-name")` accessor. Let's have a look at the most common accessors to interact with forms.
```

{{<highlight html>}}
<input type="email" name="mail-input" />
{{</highlight>}}

{{<highlight javascript>}}
await _setValue(_emailbox('mail-input'), 'do-not-spam@sakuli.io')
{{</highlight>}}

### Passwords
Password fields are a little different compared to standard input fields as they are hiding the provided input. Sakuli provides a [_password() accessor](https://sakuli.io/apidoc/sakuli-legacy/interfaces/accessorapi.html#_password) to access `<input>` fields with `type=password`. In combination with the [_setValue() action](https://sakuli.io/apidoc/sakuli-legacy/interfaces/actionapi.html#_setvalue), it is easily possible to set passwords in those fields.

{{<highlight html>}}
<input type="password" name="password-input" />
{{</highlight>}}

{{<highlight javascript>}}
await _setValue(_password('password-input'), "$ecret")
{{</highlight>}}

## Buttons
### Standard buttons

> Just as for other elements, Sakuli provides a [_button accessor](https://sakuli.io/apidoc/sakuli-legacy/interfaces/accessorapi.html#_button). The difference to e.g. input fields is the action you perform on button elements. Instead of setting a text value you would rather click the button. To do so, Sakuli provides a [_click action](https://sakuli.io/apidoc/sakuli-legacy/interfaces/actionapi.html#_click).

Reviewer: ```suggestion
Just like other elements, Sakuli provides a [_button accessor](https://sakuli.io/apidoc/sakuli-legacy/interfaces/accessorapi.html#_button). The difference to e.g. input fields is the action you perform on button elements. Instead of setting a text value, you would rather click the button. To do so, Sakuli provides a [_click action](https://sakuli.io/apidoc/sakuli-legacy/interfaces/actionapi.html#_click).
```
